packages/application-shell/src/components/quick-access/butler/butler.js

```diff
@@ -249,37 +249,34 @@ export class Butler extends React.Component {
           // the complete text is selected (when browsing through history)
           if (!isCursorAtEnd && !isEverythingSelected) return null;
 
-          this.props.getNextCommands(command).then(
-            nextCommands => {
-              this.setState({ hasNetworkError: false });
-              if (nextCommands.length > 0) {
-                // avoid moving cursor when there are sub-options
-                this.setState(prev =>
-                  // Ensure the search text has not changed while we were loading
-                  // the next results, otherwise we'd interrupt the user.
-                  // Throw away the results in case the search text has changed.
-                  prev.searchText === searchText
-                    ? {
-                        stack: [
-                          ...prevState.stack,
-                          {
-                            searchText: prevState.searchText,
-                            results: prevState.results,
-                            selectedResult: prevState.selectedResult,
-                          },
-                        ],
-                        selectedResult: 0,
-                        enableHistory: false,
-                        results: nextCommands || [],
-                      }
-                    : null
-                );
-              } else {
-                this.shake();
-              }
-            },
-            () => this.setNetworkError()
-          );
+          const nextCommands = this.props.getNextCommands(command);
```

Comment:
This is not sync, is it?

Reply:
Yes it's sync. The commands return a plain object or a list of objects. Never a promise.